examples/product-recommender/lancedb_cloud/main.py

```diff
@@ -0,0 +1,137 @@
+    'products.csv.zip',
+    'orders.csv.zip'
+]
+
+if __name__ == "__main__":
+    args = args_parse()
+    for filename in files:
+        with zipfile.ZipFile(filename, 'r') as zip_ref:
+            zip_ref.extractall('./')
+
+    products = pd.read_csv('products.csv')
+    orders = pd.read_csv('orders.csv')
+    order_products = pd.concat([pd.read_csv('order_products__train.csv'), pd.read_csv('order_products__prior.csv')])
+
+    customer_order_products = pd.merge(orders, order_products, how='inner',on='order_id')
+
+    # create confidence table
+    data = customer_order_products.groupby(['user_id', 'product_id'])[['order_id']].count().reset_index()
+    data.columns=["user_id", "product_id", "total_orders"]
+    data.product_id = data.product_id.astype('int64')
+
+    data_new = pd.DataFrame([[data.user_id.max() + 1, 46149, 50], # user 1 orders 50 Zero Calorie Cola 
+                            [data.user_id.max() + 2, 27845, 49], # user 2 orders 49 Organic Whole Milk
+                            [data.user_id.max() + 2, 26604, 32] # user 2 orders 32 Organic Blackberries
+                            ], columns=['user_id', 'product_id', 'total_orders'])
+    data = pd.concat([data, data_new]).reset_index(drop = True)
+
+    # extract unique user and product ids
+    unique_users = list(np.sort(data.user_id.unique()))
+    unique_products = list(np.sort(products.product_id.unique()))
+    purchases = list(data.total_orders)
+
+    # create zero-based index position <-> user/item ID mappings
+    index_to_user = pd.Series(unique_users)
+
+    # create reverse mappings from user/item ID to index positions
+    user_to_index = pd.Series(data=index_to_user.index + 1, index=index_to_user.values)
+
+    # create row and column for user and product ids
+    users_rows = data.user_id.astype(int)
+    products_cols = data.product_id.astype(int)
+
+    # create CSR matrix
+    matrix = scipy.sparse.csr_matrix((purchases, (users_rows, products_cols)), shape=(len(unique_users) + 1, len(unique_products) + 1))
+    matrix.data = np.nan_to_num(matrix.data, copy=False)
+
+    #split data into train and test splits
+    train, test = evaluation.train_test_split(matrix, train_percentage=0.9)
+
+    # initialize the recommender model
+    model = implicit.als.AlternatingLeastSquares(factors=args.factors,
+                                                regularization=args.regularization,
+                                                iterations=args.iterations,
+                                                num_threads=args.num_threads)
+
+    alpha = 15
+    train = (train * alpha).astype('double')
+
+    # train the model on CSR matrix
+    model.fit(train, show_progress = True)
+
+    test = (test * alpha).astype('double')
+    evaluation.ranking_metrics_at_k(model, train, test, K=100,
+                            show_progress=True, num_threads=1)
+    
+
+    db_url = "your-project-name"
+    api_key="sk_..."
+    region = "us-east-1-dev"
```

Comment:
nit - not using dev environment
```suggestion
    region = "us-east-1"
```